src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -233,6 +248,17 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         event_mask |= COR_PRF_DISABLE_OPTIMIZATIONS;
     }
 
+    if (!is_calltarget_enabled || !is_net46_or_greater || !IsNGENEnabled())
+    {
+        Info("NGEN is disabled.");
+        event_mask |= COR_PRF_DISABLE_ALL_NGEN_IMAGES;
+    }
+    else
+    {
+        Info("NGEN is enabled.");
+        event_mask |= COR_PRF_MONITOR_CACHE_SEARCHES;
+    }
```

Comment:
not a blocker, but I find it easier to understand the logic if we reverse it:

```suggestion
    if (is_calltarget_enabled && is_net46_or_greater && IsNGENEnabled())
    {
        Info("NGEN is enabled.");
        event_mask |= COR_PRF_MONITOR_CACHE_SEARCHES;
    }
    else
    {
        Info("NGEN is disabled.");
        event_mask |= COR_PRF_DISABLE_ALL_NGEN_IMAGES;
    }
```